cmd/query/app/server.go

```diff
@@ -79,14 +81,35 @@ func createHTTPServer(querySvc *querysvc.QueryService, queryOpts *QueryOptions,
 
 	apiHandler.RegisterRoutes(r)
 	RegisterStaticHandler(r, logger, queryOpts)
-	compressHandler := handlers.CompressHandler(r)
+	var compressHandler http.Handler
+	if queryOpts.BearerTokenPropagation {
+		compressHandler = handlers.CompressHandler(BearTokenPropagationHandler(logger, r))
+	} else {
+		compressHandler = handlers.CompressHandler(r)
+	}
 	recoveryHandler := recoveryhandler.NewRecoveryHandler(logger, true)
-
 	return &http.Server{
 		Handler: recoveryHandler(compressHandler),
 	}
 }
 
+func BearTokenPropagationHandler(logger *zap.Logger, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		authHeaderValue := r.Header.Get("Authorization")
+		if authHeaderValue != "" {
+			bearerToken := strings.Split(authHeaderValue, " ")
+			if len(bearerToken) == 2 {
+				ctx = context.WithValue(ctx, "Storage-Token", bearerToken)
```

Comment:
The recommended way of storing values in Context is by using private keys:
```go
type storageContextKeyType string
var storageContextKey = storageContextKeyType("Storage-Token")
```

Reply:
Thanks for the tip, I'll change it